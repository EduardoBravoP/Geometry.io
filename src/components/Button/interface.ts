import { TouchableOpacityProps } from "react-native";

export interface ButtonComponentProps extends TouchableOpacityProps{
  children: string;
}