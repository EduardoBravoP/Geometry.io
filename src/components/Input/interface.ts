import { TextInputProps } from "react-native";

export interface InputProps extends TextInputProps {
  label: string;
  placeholder: string;
  value: string;
  onChangeText: (value: string) => void
  secretInput?: boolean;
}