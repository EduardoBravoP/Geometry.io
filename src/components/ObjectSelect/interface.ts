export interface ObjectSelectProps {
  objectNumber: number;
}