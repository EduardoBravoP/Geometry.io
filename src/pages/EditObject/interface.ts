import { StackScreenProps } from '@react-navigation/stack';
import { SettingsStackParamList } from '../../@types/navigation';

export type EditObjectProps = StackScreenProps<SettingsStackParamList, 'EditObject'>;

export interface ColorProps {
  color: string;
}