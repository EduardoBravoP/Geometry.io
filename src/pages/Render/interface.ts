import { ObjectModel } from "../../common/models/object.model";

export interface ObjectSettingsProps {
  object1: ObjectModel
  object2: ObjectModel
  object3: ObjectModel
}