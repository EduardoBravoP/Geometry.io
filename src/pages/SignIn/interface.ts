export interface SignInViewModelProps {
  email: string;
  password: string;
  setEmail: React.Dispatch<React.SetStateAction<string>>;
  setPassword: React.Dispatch<React.SetStateAction<string>>;
  isLoading: boolean;
  onSubmit: () => void;
}