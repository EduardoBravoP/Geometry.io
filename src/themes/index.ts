export default {
  COLORS: {
    WHITE: '#FFFFFF',

    BACKGROUND: '#22272C',
    
    GRAY_500: '#383D41',
    GRAY_400: '#7A7D80',
    GRAY_100: '#D3D3D3',

    RED_500: '#DA0000',

    BLUE_500: '#3983DA'
  },
  FONT_FAMILY: {
    REGULAR: 'Montserrat_400Regular',
    MEDIUM: 'Montserrat_500Medium',
    SEMI_BOLD: 'Montserrat_600SemiBold'
  },
  FONT_SIZE: {
    XS: '12px',
    SM: '14px',
    MD: '16px',
    XL: '32px'
  }
}