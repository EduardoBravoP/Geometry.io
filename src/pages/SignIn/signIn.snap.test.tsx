import { ThemeProvider } from "styled-components/native";
import { SignInViewModelProps } from "./interface"
import SignInView from "./view"
import renderer from 'react-test-renderer';
import themes from "../../themes";

jest.mock('./view.model', () => {
  const fnViewModel = jest.fn()

  fnViewModel.mockImplementation(() => {
    return {
      email: 'test@test.com',
      isLoading: false,
      onSubmit: jest.fn(),
      password: '123123',
      setEmail: jest.fn(),
      setPassword: jest.fn()
    } as SignInViewModelProps
  })

  return fnViewModel
})

describe('<SignIn /> page snapshots', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider theme={themes}>
        <SignInView />
      </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})