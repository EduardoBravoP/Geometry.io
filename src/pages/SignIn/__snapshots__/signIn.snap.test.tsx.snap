// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> page snapshots should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#22272C",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    keyboardShouldPersistTaps="handled"
    style={
      {
        "backgroundColor": "#22272C",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
      }
    }
  >
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 32,
            "marginTop": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat_600SemiBold",
              "fontSize": 32,
            }
          }
        >
          Geometry.
        </Text>
        <Text
          red={true}
          style={
            {
              "color": "#DA0000",
              "fontFamily": "Montserrat_600SemiBold",
              "fontSize": 32,
            }
          }
        >
          io
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat_600SemiBold",
            "fontSize": 32,
          }
        }
      >
        Entrar
      </Text>
      <Text
        style={
          {
            "color": "#7A7D80",
            "fontFamily": "Montserrat_400Regular",
            "fontSize": 14,
            "marginBottom": 32,
            "marginTop": 8,
          }
        }
      >
        Bem vindo de volta!
      </Text>
      <View
        style={
          {
            "gap": 16,
          }
        }
      >
        <View
          style={{}}
        >
          <Text
            style={
              {
                "color": "#7A7D80",
                "fontFamily": "Montserrat_400Regular",
                "fontSize": 12,
                "marginBottom": 4,
              }
            }
          >
            E-mail
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomLeftRadius": 12,
                "borderBottomRightRadius": 12,
                "borderColor": "#383D41",
                "borderTopLeftRadius": 12,
                "borderTopRightRadius": 12,
                "borderWidth": 1,
                "flexDirection": "row",
                "gap": 16,
                "justifyContent": "space-between",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              keyboardType="email-address"
              onChangeText={[MockFunction]}
              placeholder="Insira seu e-mail"
              placeholderTextColor="#D3D3D3"
              secureTextEntry={false}
              style={
                {
                  "color": "#FFFFFF",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontFamily": "Montserrat_400Regular",
                  "fontSize": 14,
                  "paddingBottom": 18,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 18,
                }
              }
              value="test@test.com"
            />
          </View>
        </View>
        <View
          style={{}}
        >
          <Text
            style={
              {
                "color": "#7A7D80",
                "fontFamily": "Montserrat_400Regular",
                "fontSize": 12,
                "marginBottom": 4,
              }
            }
          >
            Senha
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomLeftRadius": 12,
                "borderBottomRightRadius": 12,
                "borderColor": "#383D41",
                "borderTopLeftRadius": 12,
                "borderTopRightRadius": 12,
                "borderWidth": 1,
                "flexDirection": "row",
                "gap": 16,
                "justifyContent": "space-between",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              onChangeText={[MockFunction]}
              placeholder="Insira sua senha"
              placeholderTextColor="#D3D3D3"
              secureTextEntry={true}
              style={
                {
                  "color": "#FFFFFF",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontFamily": "Montserrat_400Regular",
                  "fontSize": 14,
                  "paddingBottom": 18,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 18,
                }
              }
              value="123123"
            />
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                  "paddingRight": 16,
                }
              }
              testID="showPasswordButton"
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#3983DA",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 32,
            "opacity": 1,
            "paddingBottom": 18,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 18,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat_600SemiBold",
              "fontSize": 16,
            }
          }
        >
          Entrar
        </Text>
        <Text />
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
